libraries/message/src/descriptor.rs

```diff
@@ -70,6 +70,15 @@ pub struct Node {
 
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub path: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub git: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub branch: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub tag: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub rev: Option<String>,
```

Comment:
What if there is multiple one that does not make sense?

Reply:
Only one of them is allowed. This is checked here: https://github.com/dora-rs/dora/blob/20a1ade6add4d22ea7c9324aca87ea96651f9f91/libraries/core/src/descriptor/mod.rs#L168